caffe2/operators/reduce_ops.cc

```diff
@@ -91,12 +93,11 @@ bool ReduceMeanOp<T, Context>::Compute(
     const TIndex X_size,
     vector<TIndex>& dims,
     T* Y_data,
+    const TIndex Y_size,
     vector<int>& axes,
     int keepdims) {
-  math::Set<T, Context>(X_size, 0.f, Y_data, &context_);
-  ComputeSum<T, Context>(X_data, X_size, dims, Y_data, axes, keepdims);
-  Output(0)->Resize(dims);
-  TIndex Y_size = Output(0)->size();
+  math::Set<T, Context>(Y_size, 0.f, Y_data, &context_);
+  ComputeOp<T, Context>(X_data, X_size, dims, Y_data, axes, keepdims, Add);
   math::Scale(
       Y_size, static_cast<T>(Y_size) / X_size, Y_data, Y_data, &context_);
```

Comment:
Just checked the code, you should not cast the second parameter of math::Scale to T (although we may only support float case), alpha must be float.